### PR TITLE
[SPARK-42144][CORE][SQL] Handle null string values in StageDataWrapper/StreamBlockData/StreamingQueryData

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Similar to #39666, this PR handles null string values in StageDataWrapper/StreamBlockData/StreamingQueryData

### Why are the changes needed?
Properly handles null string values in the protobuf serializer.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
New UTs

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @gengliangwang

File: core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto [hunk 1]
```diff
@@ -495,9 +495,10 @@ message RDDOperationGraphWrapper {
 }
 
 message StreamingQueryData {
-  string name = 1;
+  optional string name = 1;
+  // convert from uuid, id should not be null
```
Comment (gengliangwang) [inline core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:499]: let's simply convert all the string fields. We can even have tests to check if there is any string which is not optional.

Comment (LuciferYang) [reply core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto:499]: OK

**Review by gengliangwang — APPROVED** [review]

**Comment by LuciferYang** [comment]
rebased

**Comment by LuciferYang** [comment]
GA passed

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Comment by LuciferYang** [comment]
thanks @gengliangwang

**Pull request closed without merging**